### PR TITLE
feat(mzData): read only data from v1.05

### PR DESCRIPTION
**Pull request by maasencioh**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export function decoder(encoded, options) {
+    var buffer = toByteArray(encoded);
+    var float;
+    if (options.precision === '64') {
+        float = new Float64Array(buffer.buffer);
+    } else {
+        float = new Float32Array(buffer.buffer);
+    }
+    return Array.from(float);
```
Comment (targos) [inline src/utils.js]: why not return the typed array?

**Comment by targos** [comment]
Can you put the test files outside of `src`? we don't want them on npm.

File: src/__tests__/mzData.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {mzData} from '..';
+import {readFileSync} from 'fs';
+const pathFiles = __dirname + '/files/';
+
+describe('mzData', () => {
+    it('read tiny.mzData.xml', () => {
+        const data = readFileSync(pathFiles + 'tiny.mzData.xml');
+        var response = mzData(data);
+        expect(response.kind).toBe('mzData');
```
Comment (targos) [inline src/__tests__/mzData.js]: what does it mean?

File: src/mzData.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import {parser} from 'sax';
+import {decoder, mergeSeries} from './utils';
+
+const xml = parser(true, {trim: true});
```
Comment (targos) [inline src/mzData.js]: it seems wrong to create the parser outside of the function. is it documented to do so?

File: src/__tests__/mzData.js (hunk 2 above)
Comment (maasencioh) [inline src/__tests__/mzData.js]: there are several related file formats, like mzML and mzXML, so it's a way to distinguish them

Comment (targos) [inline src/__tests__/mzData.js]: but you just called `mzData()` so you know what it is in advance, right?

File: src/utils.js (hunk 1 above)
Comment (maasencioh) [inline src/utils.js]: Currently is not supported on chromatogram. But indeed, in the merge step I could use them as that

File: src/mzData.js (hunk 3 above)
Comment (maasencioh) [inline src/mzData.js]: no, but indeed sounds correct what you said

File: src/__tests__/mzData.js (hunk 2 above)
Comment (maasencioh) [inline src/__tests__/mzData.js]: ok indeed, the package is not meant to do an evaluation of this, so I can remove it

File: src/utils.js [hunk 4]
```diff
@@ -2,17 +2,15 @@ import {toByteArray} from 'base64-js';
...
     } else {
-        float = new Float32Array(buffer.buffer);
+        return new Float32Array(buffer.buffer);
     }
-    return Array.from(float);
 }
 
 export function mergeSeries(first, second) {
-    var ans = first.slice();
+    var ans = new Array(first.length);
```
Comment (targos) [inline src/utils.js]: much better :)

Comment (maasencioh) [inline src/utils.js]: is that a LGTM?

Comment (targos) [inline src/utils.js]: on this line, yes

**Pull request merged**